corehq/apps/data_pipeline_audit/management/commands/find_sql_cases_not_updated_in_es.py

```diff
@@ -0,0 +1,103 @@
+from collections import namedtuple
+
+from django.core.management.base import BaseCommand
+
+from corehq.form_processor.backends.sql.dbaccessors import CaseReindexAccessor, \
+    iter_all_ids_chunked, CaseAccessorSQL
+from corehq.form_processor.change_publishers import publish_case_saved
+from corehq.sql_db.util import get_db_aliases_for_partitioned_query
+from corehq.util.argparse_types import date_type
+
+from corehq.apps.es import CaseES
+from corehq.form_processor.models import CommCareCaseSQL
+from corehq.util.dates import iso_string_to_datetime
+
+
+class Command(BaseCommand):
+    help = "Print IDs and info of sql cases whose server_modified_on is wrong in ES."
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '-d',
+            '--domain',
+            dest='domain',
+            help="Filter by this domain (optional)",
+        )
+        parser.add_argument(
+            '-s',
+            '--startdate',
+            dest='start',
+            type=date_type,
+            help="The start date. Only applicable to case on SQL domains. - format YYYY-MM-DD",
+        )
+        parser.add_argument(
+            '-e',
+            '--enddate',
+            dest='end',
+            type=date_type,
+            help="The end date. Only applicable to case on SQL domains. - format YYYY-MM-DD",
+        )
+        parser.add_argument(
+            '--fix',
+            dest='fix',
+            action='store_true',
+            help="Whether to fix by republishing to kafka.",
+            default=False,
+        )
+
+    def handle(self, **options):
+        domain = options.get('domain')
+        startdate = options.get('start')
+        enddate = options.get('end')
+        fix = options.get('fix')
+
+        chunked_case_ids = get_case_ids_chunked(
+            domain=domain, start_date=startdate, end_date=enddate)
+
+        print('Case ID\tLast Modified\tDomain')
+        for case_ids_chunk in chunked_case_ids:
+            es_modified_set = get_es_id_modified_list(domain, case_ids_chunk)
+
+            sql_modified_set = get_sql_id_modified_list(domain, case_ids_chunk)
+            bad_set = sql_modified_set - es_modified_set
+            for info in bad_set:
+                print(f'{info.case_id}\t{info.server_modified_on}\t{info.domain}')
+            if fix:
+                for case in CaseAccessorSQL.get_cases([info.case_id for info in bad_set]):
+                    publish_case_saved(case, send_post_save_signal=False)
+
+
+def get_case_ids_chunked(domain=None, start_date=None, end_date=None):
+    reindex_accessor = CaseReindexAccessor(
+        domain=domain, start_date=start_date, end_date=end_date)
+    yield from iter_all_ids_chunked(reindex_accessor)
```

Comment:
These queries take a really long time for some reason. The initial chunk takes about 30 seconds and (maybe sometimes a lot longer?) to fetch 500 cases, which seems really slow to me. The query I tried was `get_case_ids_chunked(domain=None, start_date=date(2020, 2, 1), end_date=date(2020, 3, 31))`.  @snopoke have you found `CaseReindexAccessor` to be slow in general, or might I be doing something especially wrong here?